### Statistical Data Analysis

#### Load data

In [1]:
# Import the libraries
import pandas as pd

In [2]:
# Get the dataframe
df = pd.read_csv("E:\\RajaRajeshwari\\MyFolders\\Projects\\mlops_salary_prediction\\mlops_salary_prediction_main\\data\\raw\\salary_dataset.csv", sep=",", encoding="utf-8")

In [3]:
# Get the shape of the dataframe
df.shape

(375, 6)

In [4]:
# Display the headers
df.columns

Index(['Age', 'Gender', 'Education Level', 'Job Title', 'Years of Experience',
       'Salary'],
      dtype='object')

In [5]:
# Check for missing values
df.isna().sum()

Age                    2
Gender                 2
Education Level        2
Job Title              2
Years of Experience    2
Salary                 2
dtype: int64

In [6]:
# Find and print the rows with missing values
nan_values = df[df.isna().any(axis=1)]
print(nan_values)

     Age Gender Education Level Job Title  Years of Experience  Salary
172  NaN    NaN             NaN       NaN                  NaN     NaN
260  NaN    NaN             NaN       NaN                  NaN     NaN


In [7]:
# Check for duplicate rows
df.duplicated().sum()

50

In [8]:
# Display the top 5 rows of the dataframe
df.head()

Age  Gender Education Level          Job Title  Years of Experience  \
0  32.0    Male      Bachelor's  Software Engineer                  5.0   
1  28.0  Female        Master's       Data Analyst                  3.0   
2  45.0    Male             PhD     Senior Manager                 15.0   
3  36.0  Female      Bachelor's    Sales Associate                  7.0   
4  52.0    Male        Master's           Director                 20.0   

     Salary  
0   90000.0  
1   65000.0  
2  150000.0  
3   60000.0  
4  200000.0

In [9]:
# Describe the number of non-null values and the data type
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375 entries, 0 to 374
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Age                  373 non-null    float64
 1   Gender               373 non-null    object 
 2   Education Level      373 non-null    object 
 3   Job Title            373 non-null    object 
 4   Years of Experience  373 non-null    float64
 5   Salary               373 non-null    float64
dtypes: float64(3), object(3)
memory usage: 17.7+ KB


In [10]:
# Describe numerical data
df.describe().T

count           mean           std    min      25%  \
Age                  373.0      37.431635      7.069073   23.0     31.0   
Years of Experience  373.0      10.030831      6.557007    0.0      4.0   
Salary               373.0  100577.345845  48240.013482  350.0  55000.0   

                         50%       75%       max  
Age                     36.0      44.0      53.0  
Years of Experience      9.0      15.0      25.0  
Salary               95000.0  140000.0  250000.0

In [11]:
# Describe categorical statistics
df.astype('object').describe()

Age Gender Education Level              Job Title  \
count   373.0    373             373                    373   
unique   31.0      2               3                    174   
top      33.0   Male      Bachelor's  Director of Marketing   
freq     24.0    194             224                     12   

        Years of Experience   Salary  
count                 373.0    373.0  
unique                 28.0     36.0  
top                     2.0  40000.0  
freq                   31.0     31.0

##### __Inference:__
* We have 375 rows and 6 columns
* There are 2 missing values per column, but these seem to have 2 entire rows of missing values
* There are 50 Duplicate rows
* Salary Column is the TARGET
<br>
- Numerical data columns: Age, Years of Experience, Salary
- Categorical data columns: Gender, Education Level, Job Title
    - Gender: One hot encoder
    - Education: Label encoder
    - Job title: Label encoder